tests/jballerina-unit-test/src/test/resources/test-src/workers/basic-worker-actions.bal

```diff
@@ -119,6 +120,241 @@ function workerSameThreadTest() returns any {
+        error w2a = error("15");
+        error w2b = error("15");
+        sleep(10);
+        w2a = <- w1Error;
+        w2b = <- w1Error;
+        assertValueEquality("10", w2a.message());
+        assertValueEquality("11", w2b.message());
+        return error("Completed");
+    }
+
+    record {
+        error w1Error;
+        error w2Error;
+    } x = wait {w1Error, w2Error};
+    assertValueEquality("Completed", x.w1Error.message());
+    assertValueEquality("Completed", x.w2Error.message());
+    return "done";
+}
+
+function simpleSendActionXMLType() returns string {
+    @strand {thread: "parent"}
+    worker w1Xml returns xml {
+        xml w1a = xml `10`;
+        xml w1b = xml `11`;
+        w1a -> w2Xml;
+        w1b ->> w2Xml;
+        return xml `Completed`;
+    }
+
+    @strand {thread: "parent"}
+    worker w2Xml returns xml {
+        xml w2a = xml `boom`;
+        xml w2b = xml `boom`;
+        sleep(10);
+        w2a = <- w1Xml;
+        w2b = <- w1Xml;
+        assertValueEquality(xml `10`, w2a);
+        assertValueEquality(xml `11`, w2b);
+        return xml `Completed`;
+    }
+
+    record {
+        xml w1Xml;
+        xml w2Xml;
+    } x = wait {w1Xml, w2Xml};
+    assertValueEquality("Completed", x.w1Xml.toString());
+    assertValueEquality("Completed", x.w2Xml.toString());
+    return "done";
+}
+
+
+function testSimpleSendActionWithCloneableType() {
+    assertValueEquality("done", simpleSendActionWithCloneableType());
+}
+
+type IntRec readonly & record {
+                           int ID = 0;
+                       };
+
+function testSimpleSendActionReadonlyRecord() {
+    assertValueEquality("done", simpleSendActionReadonlyRecord());
+}
+
+function testSimpleSendActionXMLType() {
+    assertValueEquality("done", simpleSendActionXMLType());
+}
+
+function testSimpleSendActionErrorType() {
+    assertValueEquality("done", simpleSendActionErrorType());
+}
+
+function simpleSendActionReadonlyRecord() returns string {
+    @strand {thread: "parent"}
+    worker w1readonlyRec returns string {
+        IntRec w1a = {ID: 10};
+        IntRec w1b = {ID: 11};
+        w1a -> w2readonlyRec;
+        w1b ->> w2readonlyRec;
+        return "Completed";
+    }
+
+    @strand {thread: "parent"}
+    worker w2readonlyRec returns string {
+        IntRec w2a = {ID: 3};
+        IntRec w2b;
+        sleep(10);
+        w2a = <- w1readonlyRec;
+        w2b = <- w1readonlyRec;
+        assertValueEquality(10, w2a.ID);
+        assertValueEquality(11, w2b.ID);
+        return "Completed";
+    }
+
+    record {
+        string w1readonlyRec;
+        string w2readonlyRec;
+    } x = wait {w1readonlyRec, w2readonlyRec};
+    assertValueEquality("Completed", x.w1readonlyRec);
+    assertValueEquality("Completed", x.w2readonlyRec);
+    return "done";
+}
+
+function simpleSendActionWithMapType() returns map<anydata> {
+    @strand {thread: "parent"}
+    worker w1 returns boolean {
+        map<int> a = { "a" : 1, "b" : 2};
```

Comment:
Shall we test `map<value:Cloneable>` and `value:Cloneable[]`? We probably already have anydata tests, right?

Reply:
Fixed https://github.com/ballerina-platform/ballerina-lang/pull/30250/commits/1ee63a7c47b0cd174504e4502ed20acc07f005cc